templates/azml-app/app/generateScore.py

```diff
@@ -0,0 +1,111 @@
+        """
+            Generates a score file for the model.
+            :param model_dir_name: The name of the model directory.
+            :param pipeline_type_name: The type of the pipeline.
+            :param tokenizer_type_name: The type of the tokenizer.
+            :param model_loader_type_name: The type of the model loader.
+            :param pipeline_task_name: The task name for the pipeline.
+            :param tokenizer_path: The path to the tokenizer.
+            :param model_asset_path: The path to the safetensor files.
+        """
+        self.model_dir_name = model_dir_name
+        self.pipeline_class_name = pipeline_class_name
+        self.tokenizer_class_name = tokenizer_class_name
+        self.model_loader_class_name = model_loader_class_name
+        self.pipeline_task_name = pipeline_task_name
+        self.tokenizer_path = tokenizer_path
+        self.model_asset_path = model_asset_path
+
+        self.score_file_type = None
+
+    def generate(self):
+        print("Generating score file...")
+        if self.score_file_type.lower() == "mlflow":
+            return self.generate_score_file_mlflow()
+        else:
+            raise ValueError(f"Invalid model type {self.score_file_type}.")
+
+    def generate_score_file_mlflow(self):
+        pipeline_import_template = ""
+        pipeline_loading_template = ""
+        if self.pipeline_task_name is not None and self.pipeline_task_name != "":
+            pipeline_loading_template = \
+                PIPELINE_LOADING_TEMPLATE \
+                .replace("TASK_NAME", f"task=\"{self.pipeline_task_name}\", ")
+            if self.pipeline_class_name is None and KNOWN_TASK_PIPELINE_MAP[self.pipeline_task_name] is not None:
+                self.pipeline_class_name = KNOWN_TASK_PIPELINE_MAP[self.pipeline_task_name]
+        else:
+            pipeline_loading_template = \
+                PIPELINE_LOADING_TEMPLATE \
+                .replace("TASK_NAME", "")
+
+        if self.pipeline_class_name is not None and self.pipeline_class_name != "":
+            pipeline_import_template = \
+                IMPORT_PIPELINE_TYPE \
+                .replace("PIPELINE_TYPE", self.pipeline_class_name)
+            pipeline_loading_template = \
+                pipeline_loading_template \
+                .replace("PIPELINE_TYPE", self.pipeline_class_name)
+        template = \
+            SCORE_FILE_TEMPLATE \
+            .replace("IMPORT_PIPELINE_TYPE", pipeline_import_template) \
+            .replace("PIPELINE_LOADING", pipeline_loading_template) \
+            .replace("TOKENIZER_TYPE", self.tokenizer_class_name) \
+            .replace("MODEL_TYPE", self.model_loader_class_name) \
+            .replace("TOKENIZER_PATH", f"/.azml_model_cache/{self.model_dir_name}/{self.tokenizer_path}") \
+            .replace("SAFETENSOR_PATH", f"/.azml_model_cache/{self.model_dir_name}/{self.model_asset_path}")
+        try:
+            with open("/app/app/score.py", "w") as f:

```

Comment:
>app/app [](http://example.com/codeflow?start=24&length=7)

did you intend /app/app or is that a typo? do we have to create the internal app directory if it doesn't exist?

Reply:
i think this is just the directory that we copied things to.